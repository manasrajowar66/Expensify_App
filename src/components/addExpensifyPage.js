import React from 'react';
import { connect } from 'react-redux';
import { addExpence } from '../actions/expences';
import ExpenceForm from './expenceForm';

export class CreateExpensifyPage extends React.Component{
    onSubmit = (expence)=>{
        this.props.addExpence(expence);
        this.props.history.push('/');
    }
    render(){
        return(
            <div>
            <h1>Add Expence</h1>
            <ExpenceForm 
                onSubmit={this.onSubmit}
            />
            </div>
        )
    }
};



const mapDispatchToProps = (dispatch) => {
    return {
        addExpence:(expence)=> dispatch(addExpence(expence))
    };
}

export default connect(undefined,mapDispatchToProps)(CreateExpensifyPage);