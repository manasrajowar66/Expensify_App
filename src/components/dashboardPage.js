import React from 'react';
import { connect } from 'react-redux';
import ExpenceList from './expenceList';
import ExpenceListFilter from './expenceListFilter';
import ExpenceSummary from './expenceSummary';

export class ExpensifyDashboardPage extends React.Component{
    
    render(){
        return(
         <div>
            {!!this.props.expences.length && <ExpenceSummary />}
            <ExpenceListFilter />
            <ExpenceList />
         </div>
      ) 
    }
}

const mapStateToProps = (state)=>{
    return{
        expences:state.expences
    }
}
export default connect(mapStateToProps)(ExpensifyDashboardPage);