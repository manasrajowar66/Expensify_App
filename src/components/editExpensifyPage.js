import React from 'react';
import { connect } from 'react-redux';
import { editExpence } from '../actions/expences';
import { removeExpence } from '../actions/expences';
import ExpenceForm from './expenceForm';

export class EditExpensifyPage extends React.Component{
    onSubmit = (expence)=>{
        this.props.editExpence(expence.id,expence);
        this.props.history.push('/');
    }
    onClick = (expence)=>{
        this.props.removeExpence(expence.id);
        this.props.history.push('/');
    }
    render(){
        return (
            <div>
             <ExpenceForm 
                expence={this.props.expence}
                onSubmit={this.onSubmit}
             />
             <button onClick={()=>{
                 this.onClick(this.props.expence);
             }}>remove</button>
            </div>
        );
    }
}



const mapStateToProps = (state,props) =>{
    return{
        expence:state.expences.find((expence)=>{
            return expence.id === props.match.params.id;
        })
    };
}

const mapDispatchToProps = (dispatch)=>{
    return{
        editExpence:(id,expence)=> dispatch(editExpence(id,expence)),
        removeExpence:(id)=> dispatch(removeExpence(id))
    };
}

export default connect(mapStateToProps,mapDispatchToProps)(EditExpensifyPage);