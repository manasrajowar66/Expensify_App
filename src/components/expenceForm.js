import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
// import 'react-dates/initialize';




export default class ExpenceForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            describtion:props.expence ? props.expence.describtion : '',
            note:props.expence ? props.expence.note : '',
            amount:props.expence ? (props.expence.amount/100).toString() : '',
            createdAt:props.expence ? moment(props.expence.createdAt) : moment(),
            calenderFocused:false,
            error:''
        } 
    }
    onDescribtionChange = (e)=>{
       const describtion = e.target.value;
       this.setState(()=>({describtion }));
    }
    onNotechange = (e)=>{
        const note = e.target.value;
        this.setState(()=>({note}))
    }
    onAmountChange = (e)=>{
        const amount = e.target.value;
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/))
        this.setState(()=>({amount}));
    }
    onDateChange = (createdAt) => {
        if(createdAt)
        this.setState(() =>({createdAt}));
    }
    onFocusChange = ({focused})=> {
      this.setState(()=>({ calenderFocused:focused}));
    }
    onSubmit = (e) => {
        e.preventDefault();
        if(!this.state.describtion || !this.state.amount)
        {
            this.setState(()=>({error:'Please provide describtion and amount'}));
        }
        else
        {
            this.setState(()=>({error:''}));
            this.props.onSubmit({
                describtion:this.state.describtion,
                amount:parseFloat(this.state.amount,10)*100,
                note:this.state.note,
                createdAt:this.state.createdAt.valueOf()
            });
        }


    }
    render(){
        return (
            <div>
                
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>

                    <input 
                    name='describtion'
                    type='text' 
                    placeholder='Describtion' 
                    autoFocus
                    value={this.state.describtion} 
                    onChange={this.onDescribtionChange}
                    />

                    <input 
                    name='amount'
                    type='text' 
                    placeholder='Amount'
                    value={this.state.amount} 
                    onChange={this.onAmountChange} 
                    />
                    <SingleDatePicker 
                    date={this.state.createdAt}
                    onDateChange={this.onDateChange}
                    focused={this.state.calenderFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={()=>false}
                    />
                    <textarea 
                    name='note'
                    placeholder='Please add a note for this expence' 
                    value={this.state.note} 
                    onChange={this.onNotechange}
                    />

                    <button>Add Expence</button>

                </form>
            </div>
        );
    }
}