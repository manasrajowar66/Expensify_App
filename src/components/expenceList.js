import React from 'react';
import ExpenceListItem from './expenceListItem';
import getVisibleExpenes from '../store/visibleExpences';
import { connect } from 'react-redux';

export const ExpenceList = (props) =>(
    <div>
        {props.expences.length ===0 ? (<p>No expences avalable</p>) :
         (<div>
            <h1>Expence List</h1>
        {props.expences.map((expence)=>{
            return (
                <ExpenceListItem key={expence.id} {...expence}/>
            )
        })}
        </div>) }
    </div>
);

const mapStateToProps = (state)=>{
    return {
        expences:getVisibleExpenes(state.expences,state.filters)
    };
}
export default connect(mapStateToProps)(ExpenceList);