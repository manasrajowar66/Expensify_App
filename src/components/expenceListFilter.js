import React from 'react';
import { setTextFilter , sortByAmount , sortByDate , setEndDate , setStartDate } from '../actions/filters';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';

class ExpenceListFilter extends React.Component{
    state={
        focusedInput:null
    }
    onDatesChange = ({startDate,endDate}) =>{
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    }
    onFocusChange = (focusedInput) =>{
        this.setState(()=>({focusedInput}));
    }
    render(){
        return(
            <div>
            <input type='text' value={this.props.filters.text} onChange={(e)=>{
                this.props.dispatch(setTextFilter(e.target.value));
            }}/>
            <select
            value={this.props.filters.sortBy} 
            onChange={(e)=>{
                const val = e.target.value;
                if(val === 'date')
                this.props.dispatch(sortByDate());
                else
                this.props.dispatch(sortByAmount());
            }}>
                <option value='date'>Date</option>
                <option value='amount'>Amount</option>
            </select>
            <DateRangePicker 
                startDate={this.props.filters.startDate}
                endDate={this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.focusedInput}
                onFocusChange={this.onFocusChange}
                numberOfMonths={1}
                isOutsideRange={()=>false}
                showClearDates={true}
            />
    
            </div>
        )
    }
}


const mapStateToProps = (state) =>{
    return{
        filters:state.filters
    };
}
export default connect(mapStateToProps)(ExpenceListFilter);