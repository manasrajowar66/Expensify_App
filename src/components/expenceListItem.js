/* eslint-disable no-unused-vars */
import React from 'react';
import { removeExpence } from '../actions/expences';
import { Link } from 'react-router-dom';
import EditExpensifyPage from './editExpensifyPage';
const ExpenceListItem = ({describtion,amount,createdAt,id}) =>(
    <div>
        <Link className="link-decoration" to={`/edit/${id}`}><h2>{describtion}</h2></Link>
        {amount} - {createdAt}
    </div>
);

export default ExpenceListItem;