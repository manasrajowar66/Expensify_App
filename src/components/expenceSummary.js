import React from 'react';
import getTotalExpences from '../store/getTotalExpences';
import getVisibleExpences from '../store/visibleExpences';
import { connect } from 'react-redux';
import numeral from 'numeral';

export const ExpenceSummary = ({totalCount , totalAmount})=>{
    const expenceWord = totalCount === 1 ? 'expence' : 'expences';
    
    return(
        <p>Viewing {totalCount} {expenceWord} totaling {numeral(totalAmount).format('$0,0.00')}</p> 
    )
}

const mapStateToProps = (state)=>{
    const expences = getVisibleExpences(state.expences,state.filters);
    return {
        totalAmount: getTotalExpences(expences)/100,
        totalCount: expences.length
    };

}

export default connect(mapStateToProps)(ExpenceSummary);