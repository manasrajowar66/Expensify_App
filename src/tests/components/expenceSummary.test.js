import React from 'react';
import { shallow } from 'enzyme';
import { ExpenceSummary } from '../../components/expenceSummary';

test('should correctly render expence summary',()=>{
    const data = {
        totalCount: 2,
        totalAmount:8039
    }
    const wrraper = shallow(<ExpenceSummary {...data}/>);
    expect(wrraper).toMatchSnapshot();

}) 