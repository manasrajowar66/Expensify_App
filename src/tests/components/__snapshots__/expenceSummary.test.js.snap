// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expence summary 1`] = `
<p>
  Viewing 
  2
   
  expences
   totaling 
  $8,039.00
</p>
`;
